app/scripts/metamask-controller.js

```diff
@@ -4850,20 +4851,22 @@ export default class MetamaskController extends EventEmitter {
     transactionOptions,
     dappRequest,
   }) {
+    const networkClientId = dappRequest?.networkClientId ??
+        this.networkController.state.selectedNetworkClientId
+    const { chainId } = this.networkController.getNetworkConfigurationByNetworkClientId(networkClientId)
+
     return {
       internalAccounts: this.accountsController.listAccounts(),
       dappRequest,
-      networkClientId:
-        dappRequest?.networkClientId ??
-        this.networkController.state.selectedNetworkClientId,
+      networkClientId,
       selectedAccount: this.accountsController.getAccountByAddress(
         transactionParams.from,
       ),
       transactionController: this.txController,
       transactionOptions,
       transactionParams,
       userOperationController: this.userOperationController,
-      chainId: getCurrentChainId({ metamask: this.networkController.state }),
+      chainId,
```

Comment:
Fixes the globally selected chainId always being passed in. Now gets the chainId for the networkClientId on the request